src/Components/Facility/ConsultationDetails/Events/EventsList.tsx

```diff
@@ -0,0 +1,79 @@
+import { useTranslation } from "react-i18next";
+import { useSlugs } from "../../../../Common/hooks/useSlug";
+import { ConsultationModel } from "../../models";
+import PaginatedList from "../../../../CAREUI/misc/PaginatedList";
+import routes from "../../../../Redux/api";
+import { TimelineNode } from "../../../../CAREUI/display/Timeline";
+import LoadingLogUpdateCard from "../../Consultations/DailyRounds/LoadingCard";
+import GenericEvent from "./GenericEvent";
+import { EventGeneric } from "./types";
+import { getEventIcon } from "./iconMap";
+
+interface Props {
+  consultation: ConsultationModel;
+}
+
+export default function EventsList({ consultation }: Props) {
+  const [consultationId] = useSlugs("consultation");
+  const { t } = useTranslation();
+
+  console.log("consultation", consultation, consultationId);
+
+  return (
+    <PaginatedList route={routes.getEvents} pathParams={{ consultationId }}>
+      {() => (
+        <>
+          <div className="mt-4 flex w-full flex-col gap-4">
+            <div className="flex max-h-[85vh] flex-col gap-4 overflow-y-auto overflow-x-hidden px-3">
+              <PaginatedList.WhenEmpty className="flex w-full justify-center border-b border-gray-200 bg-white p-5 text-center text-2xl font-bold text-gray-500">
+                <span className="flex justify-center rounded-lg bg-white p-3 text-gray-700  ">
+                  {t("no_consultation_updates")}
+                </span>
+              </PaginatedList.WhenEmpty>
+              <PaginatedList.WhenLoading>
+                <LoadingLogUpdateCard />
+              </PaginatedList.WhenLoading>
+              <PaginatedList.Items<EventGeneric> className="flex grow flex-col gap-3">
+                {(item, items) => (
+                  <TimelineNode
+                    name={
+                      item.event_type.name
+                        .split("_")
+                        .map(
+                          (text) =>
+                            text[0].toUpperCase() + text.toLowerCase().slice(1)
+                        )
+                        .join(" ") + " Event"
+                    }
+                    event={{
+                      type: item.change_type.replace(/_/g, " ").toLowerCase(),
+                      timestamp: item.created_date?.toString() ?? "",
+                      by: item.caused_by,
+                      icon: getEventIcon(item.event_type.name),
+                    }}
+                    isLast={items.indexOf(item) == items.length - 1}
+                  >
+                    {(() => {
+                      switch (item.event_type.name) {
+                        case "INTERNAL_TRANSFER":
+                        case "CLINICAL":
+                        case "DIAGNOSIS":
+                        case "ENCOUNTER_SUMMARY":
+                        case "HEALTH":
+                        default:
+                          return <GenericEvent event={item} />;
```

Comment:
Are these just here as placeholders? Just checking if this was intentional to let all these cases fall to the default case.

Reply:
It was intensional, the plan was to create a separate card for each type of event down the line